pkg/promoter/options.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package promoter
+
+import (
+	"errors"
+)
+
+// Options capture the switches available to run the image promoter
+type Options struct {
+	// Threads determines how many promotion threads will run
+	Threads int
```

Comment:
How about making this an unsigned integer?

Reply:
For now, I want to keep the interactions with the internal code just as it is in case the refactor introduces any bugs.

I've noticed several bugs while testing but I don't want to fix them either just jet as I want this to run exactly as it did before moving things around. WDYT?